### PR TITLE
Fix for ..\directory type entries for search folder

### PR DESCRIPTION
**Pull request by prawalagarwal**
Problem: The root directory path for searching test assemblies does not support ..\directory type entries. 
Fix: This fix resolves this issue.
Testing: Unit and Manual

### CODE REVIEW

File: Tasks/VsTest/vstest.ts [hunk 1]
```diff
@@ -930,4 +930,6 @@ function isNullOrWhitespace(input) {
         return true;
     }
     return input.replace(/\s/g, '').length < 1;
-}
\ No newline at end of file
+}
+
+export {isNullOrWhitespace};
```
Comment (RanjanarMS) [inline Tasks/VsTest/vstest.ts]: Move to helper class as mentioned above

File: Tests-Legacy/L0/VsTest/_suite.ts [hunk 2]
```diff
@@ -1069,4 +1069,154 @@ describe('VsTest Suite', function () {
...
+    it('Vstest task with custom adapter path with double dots is supported', (done) => {
+
+        setResponseFile('vstestGood.json');
+
+        const tr = new trm.TaskRunner('VSTest');
+        tr.setInput('testSelector', 'testAssemblies');
+        tr.setInput('testAssemblyVer2', '/source/dir/someFile1');
+        tr.setInput('vstestLocationMethod', 'version');
+        tr.setInput('vsTestVersion', '14.0');
+        tr.setInput('pathtoCustomTestAdapters', 'E:/path/to/../customadapters');
```
Comment (RanjanarMS) [inline Tests-Legacy/L0/VsTest/_suite.ts]: Can you add one test case for netwok path in all the scenarios?

File: Tasks/VsTest/taskinputparser.ts [hunk 3]
```diff
@@ -97,12 +98,20 @@ function initTestConfigurations(testConfiguration: models.TestConfigurations) {
     tl._writeLine(tl.loc('testSelectorInput', testConfiguration.testSelection));
 
     testConfiguration.testDropLocation = tl.getInput('searchFolder');
+    if (!isNullOrWhitespace(testConfiguration.testDropLocation))
```
Comment (prawalagarwal) [inline Tasks/VsTest/taskinputparser.ts]: We want don't want to change if it is null or empty. These cases are handled when they are actually used.

File: Tasks/VsTest/taskinputparser.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ import * as utils from './helpers';
 import * as os from 'os';
 import * as versionFinder from './versionfinder';
 const uuid = require('node-uuid');
+import {isNullOrWhitespace} from './vstest'
```
Comment (prawalagarwal) [inline Tasks/VsTest/taskinputparser.ts]: Yes. Moved

File: Tasks/VsTest/vstest.ts (hunk 1 above)
Comment (prawalagarwal) [inline Tasks/VsTest/vstest.ts]: Done

File: Tests-Legacy/L0/VsTest/_suite.ts (hunk 2 above)
Comment (prawalagarwal) [inline Tests-Legacy/L0/VsTest/_suite.ts]: Will discuss in scrum

**Pull request merged**